appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/eesecurity/SignedJWTIdentityStore.java

```diff
@@ -214,8 +223,12 @@ public CredentialValidationResult validate(SignedJWTCredential signedJWTCredenti
             return Optional.empty();
         }
 
-        try (InputStream inputStream = publicKeyURL.openStream();
-             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))){
+        URLConnection urlConnection = publicKeyURL.openConnection();
+        ContentType contentType = ContentType.newContentType(urlConnection.getContentType());
+
+        Charset charset = contentType.getCharacterEncoding() != null ? Charset.forName(contentType.getCharacterEncoding()) : Charset.defaultCharset();
```

Comment:
You might want to check if the charset from the contentType is supported besides if the contentType contains one. Otherwise you might get an UnsupportedCharsetException here.

Reply:
Agreed